numpy/lib/function_base.py

```diff
@@ -1891,30 +1896,33 @@ def diff(a, n=1, axis=-1):
     >>> np.diff(x, axis=0)
     array([[-1,  2,  0, -2]])
 
+    >>> x = np.arange('1066-10-13', 3, dtype=np.datetime64)
```

Comment:
This does look rather weird. Even though it works, I would suggest to replace it with something that would more directly make sense to a user:
```
np.arange('1066-10-13', '1066-10-16', dtype=np.datetime64)
```